rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp

```diff
@@ -0,0 +1,419 @@
+    "Alpha", 1.0, "0 is fully transparent, 1.0 is fully opaque.",
+    this, SLOT(updateColorAndAlpha()));
+
+  width_property_ = new rviz_common::properties::FloatProperty(
+    "Width", 0.02, "Width to drow effort circle", this,
+    SLOT(updateColorAndAlpha()));
+
+  scale_property_ = new rviz_common::properties::FloatProperty(
+    "Scale", 1.0, "Scale to draw effort circle", this,
+    SLOT(updateColorAndAlpha()));
+
+  history_length_property_ = new rviz_common::properties::IntProperty(
+    "History Length", 1, "Number of prior measurements to display.", this,
+    SLOT(updateHistoryLength()));
+
+  history_length_property_->setMin(1);
+  history_length_property_->setMax(100000);
+
+  robot_description_property_ =
+    new rviz_common::properties::StringProperty(
+    "Robot Description", "/robot_description",
+    "Name of the topic from which to load the robot "
+    "description.",
+    this, SLOT(updateRobotDescription()));
+
+  tf_prefix_property_ = new rviz_common::properties::StringProperty(
+    "TF Prefix", "",
+    "Robot Model normally assumes the link name is the same as the tf frame name. "
+    "This option allows you to set a prefix.  Mainly useful for multi-robot situations.",
+    this, SLOT(updateTfPrefix()));
+
+  joints_category_ = new rviz_common::properties::Property("Joints", QVariant(), "", this);
+}
+
+EffortDisplay::~EffortDisplay()
+{
+  for (auto & j : this->joints_) {
+    free(j.second);
+  }
+}
+
+void EffortDisplay::onInitialize()
+{
+  MFDClass::onInitialize();
+  updateHistoryLength();
+}
+
+// Set the number of past visuals to show.
+void EffortDisplay::updateHistoryLength()
+{
+  while (visuals_.size() > static_cast<size_t>(history_length_property_->getInt())) {
+    visuals_.pop_front();
+  }
+}
+
+// Clear the visuals by deleting their objects.
+void EffortDisplay::reset()
+{
+  MFDClass::reset();
+  visuals_.clear();
+}
+
+void EffortDisplay::load(const rviz_common::Config & config)
+{
+  rviz_common::Display::load(config);
+}
+
+void EffortDisplay::topic_callback(const std_msgs::msg::String & msg)
+{
+  robot_description_ = msg.data;
+  robot_model_ = std::shared_ptr<urdf::Model>(new urdf::Model());
+  if (!robot_model_->initString(robot_description_)) {
+    // ROS_ERROR("Unable to parse URDF description!");
+    setStatus(
+      rviz_common::properties::StatusProperty::Error,
+      "URDF", "Unable to parse robot model description!");
+    return;
+  }
+  setStatus(rviz_common::properties::StatusProperty::Ok, "URDF", "Robot model parsed Ok");
+
+  for (std::map<std::string, urdf::JointSharedPtr>::iterator it = robot_model_->joints_.begin();
+    it != robot_model_->joints_.end(); it++)
+  {
+    urdf::JointSharedPtr joint = it->second;
+    if (joint->type == urdf::Joint::REVOLUTE || joint->type == 2) {
+      std::string joint_name = it->first;
+      urdf::JointLimitsSharedPtr limit = joint->limits;
+      joints_[joint_name] = createJoint(joint_name);
```

Comment:
At this point, `createJoint` is a one-liner, so I'll suggest we just inline it here:

```suggestion
      joints_[joint_name] = new JointInfo(joint_name, joints_category_);
```

and remove the the `createJoint` method.